var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -276,3 +276,7 @@ def cmake_args(self):
         options.append('-DBUILD_SHARED_LIBS=%s' % ('+shared' in self.spec))
 
         return options
+
+    def install(self, spec, prefix):
```

Comment:
@eugeneswalker this looks like what I expected (it might have required a call to `super(... ).install()`, but since this isn't being encountered at all that's not the problem here), so I'll have to investigate over the next couple days why this phase is never executed (since it is a listed phase of all CMake-based packages).

FWIW this approach would definitely "work" if we placed the check in `cmake_args` (since that function is definitely getting called). We may even do that in a couple packages with checks like this; I'll also be looking into that.

Reply:
@eugeneswalker after looking at it more I see what I missed: `CMakePackage` packages define 3 phases: `cmake`, `build`, and `install`; they run in that order. https://github.com/spack/spack/issues/20347 shows that the error occurs during the `cmake` phase (so performing a check in the install phase will not catch the problem in time).

For this approach to work, the check would have to move into the `cmake` phase. We can achieve that by performing the check at the beginning of `cmake_args` in the `kokkos` package definition. Would you be able to try that?

Reply:
> For this approach to work, the check would have to move into the `cmake` phase. We can achieve that by performing the check at the beginning of `cmake_args` in the `kokkos` package definition. Would you be able to try that?

Done. Just pushed.